### ISSUE TITLE
Allow to use pnpm as a publish utility

### ISSUE THREAD

**Issue by stalniy** [semantic-release/npm#280]
## The issue

I use [pnpm](https://pnpm.js.org/) for monorepo management and semantic-release for publishing new releases. 
[pnpm supports `workspace:` protocol](https://pnpm.js.org/en/workspaces#workspace-ranges-workspace) in package.json, so the packages in monorepo can be always linked to local versions, during `publish` phase this `woskpace:` protocol is removed. 

**The issue** is that semantic release uses `npm` cli tool for publishing which does not remove `workspace:` protocol and eventually the package is published with `workspace:` dependency which cannot be resolved

## Desired behavior

Allow to specify either the string template for publishing package or function in plugin options

## Workaround

In linux, I can use aliases and alias `npm` to `pnpm`

**Comment by stalniy** [comment]
From the source code, it looks like [this the line](https://github.com/semantic-release/npm/blob/master/lib/publish.js#L25) that needs to be configured

**Comment by stalniy** [comment]
I haven't created PR yet because there was no approval from core contributors of this repo. Usually, things don't get merged without previous discussion :)

**Comment by bestickley** [comment]
Bummer! Thanks for letting me know.

**Comment by travi** [comment]
> because there was no approval from core contributors of this repo

@stalniy we do appreciate the consideration for that. We do prefer having some agreement before you invest effort into a PR

Apologies for the delayed response. Our maintainers team had been very busy, so some issues like this one have not gotten attention. 

It sounds like this is only a need for projects structured as monorepos, is that correct @stalniy @bestickley? semantic-release does not provide official support for monorepos at this time and it is unlikely that we will be able to provide official support

**Comment by bestickley** [comment]
@travi, yes that is my use. I would like to run semantic-release recursively throughout my monorepo with pnpm. Totally understand supporting monorepos adds complication to maintaining. From first glance it seems simple to me, but I totally respect your decision as you have a broader perspective. Appreciate your work on this awesome project!

**Comment by stalniy** [comment]
would be awesome to have an extension point in the context of this package instead of making a completely new one that copies everything except of few lines

**Comment by NullVoxPopuli** [comment]
Just ran in to this. would be great to have fixed.

**Comment by VanishMax** [comment]
I support having this feature implemented. Currently, I always run into NPM errors because of the dependencies installed by PNPM

**Comment by WazzaJB** [comment]
After struggling to get this to work I ended up just removing `@semantic-release/npm` and running `pnpm -r publish` after multi-semantic-release had run, seems to hold up okay although I am sure there will be caveats to this approach.

**Comment by andorfermichael** [comment]
PNPM is well established and should therefore definitely be supported

**Comment by BinToss** [comment]
> After struggling to get this to work I ended up just removing `@semantic-release/npm` and running `pnpm -r publish` after multi-semantic-release had run, seems to hold up okay although I am sure there will be caveats to this approach.

I did something similar for NuGet. The caveat is: Semantic-Release exits with code 0 if no release is necessary. However, Semantic Release still ran the `dotnet pack` command (configured via semantic-release/exec's prepareCmd). This produced a package with an existing release's version, so `dotnet nuget push *.nupkg` will exit with an error.

I have a feeling PNPM would experience a similar issue.

**Comment by prisis** [comment]
Hey :), if you still looking for a pnpm solution check https://github.com/anolilab/semantic-release/blob/main/packages/semantic-release-pnpm/README.md

**Comment by tylerlaws0n** [comment]
another option for those here looking:

```js
    [
      '@semantic-release/exec',
      {
        prepareCmd: 'pnpm version ${nextRelease.version} --git-tag-version=false',
        publishCmd: 'pnpm publish --no-git-checks',
      },
    ],
```

**Comment by difosfor** [comment]
> It sounds like this is only a need for projects structured as monorepos, is that correct [@stalniy](https://github.com/stalniy) [@bestickley](https://github.com/bestickley)? semantic-release does not provide official support for monorepos at this time and it is unlikely that we will be able to provide official support any time soon. The most likely improvement we might provide is to improve support for tools built on top of semantic-release. Could you describe your use case so we can consider it if we do invest in that at some point?

We also use pnpm for our non-mono semantic-release repos. It seems like something that deserves official plugin support to me; at least for non-mono repos. Similarly for other popular package managers like yarn and bun.

In case anybody else is interested I'm currently hacking around the lack of pnpm and Trusted Publishing support with the following config:
```
{
  plugins: [
    '@semantic-release/commit-analyzer',
    '@semantic-release/release-notes-generator',
    '@semantic-release/changelog',
    // Lots of problems to work around here:
    // 1) There is no standard maintained (@semantic-release) pnpm plugin
    // 2) @semantic-release/npm does not support Trusted Publishing
    // 3) pnpm does not support Trusted Publishing
    // 4) Package manager is pinned to pnpm
    // So yeah, sorry for this trash pile of hacks until we get proper solutions
    [
      '@semantic-release/exec',
      {
        // biome-ignore-start lint/suspicious/noTemplateCurlyInString: supported by @semantic-release/exec
        prepareCmd:
          'pnpm version ${nextRelease.version} --no-git-checks --no-git-tag-version && ' +
          'RELEASE_VERSION=${nextRelease.version} pnpm run build:lib && ' +
          'pnpm pack',
        publishCmd:
          'cp package.json package.json.bak && ' +
          'jq "del(.devEngines.packageManager)" package.json.bak > package.json && ' +
          'npm publish liveryvideo-player-*.tgz --tag ${nextRelease.channel || "latest"} && ' +
          'mv package.json.bak package.json',
        // biome-ignore-end lint/suspicious/noTemplateCurlyInString: see above
      },
    ],
    '@semantic-release/git',
  ],
}
```
